internal/pkg/server/server.go

```diff
@@ -99,32 +101,46 @@ func (s *Server) streamImages(c echo.Context, pullAuth auth.Authenticator, image
 		return c.NoContent(http.StatusBadRequest)
 	}
 
-	tar, err := s.pullAndSaveImages(c.Request().Context(), pullAuth, images)
+	tar, err := s.pullAndSaveImages(c.Request().Context(), c, pullAuth, images)
 	if err != nil {
-		return err
+		return echo.ErrForbidden
```

Comment:
Could we check here what kind of error did happen? `s.pullAndSaveImages` can return different errors for network errors, ...

usually `errors.Is(err, ERR)` is used in go.

Reply:
I tried it. The problem is the error is a full string, which looks something like this: "status":500,"error":"Error response from daemon: Get https://pathToImage: denied: access forbidden". If I want to check what kind of error happened, I would have to use regex.

Reply:
That's no fun. If the author in the underlying library does not implement different error types it is hard to "parse" the correct error.

I'm not sure if a forbidden error when having network issues is even more confusing than a 500 if authentication fails 😅